### PR TITLE
Improve deparser expression handling, fix deparsing of DEFAULT expressions classified as "a_expr"

### PR DESCRIPTION
**Pull request by lfittl**
Alternate to #269 that fixes the discussed edge cases there as well.

Best reviewed by looking at individual commits.

### CODE REVIEW

File: test/deparse_tests.c [hunk 1]
```diff
@@ -409,7 +409,11 @@ const char* tests[] = {
   "CREATE INDEX myindex ON public.mytable USING btree (col1, (col2::varchar) varchar_pattern_ops)",
   "SELECT * FROM CAST(1 AS text)",
   "DECLARE \"Foo1\" SCROLL CURSOR FOR SELECT * FROM tenk1 ORDER BY unique2",
-  "FETCH BACKWARD 23 \"Foo1\""
+  "FETCH BACKWARD 23 \"Foo1\"",
+  "CREATE TABLE my_table (created_at timestamptz NOT NULL DEFAULT (now() AT TIME ZONE 'UTC'))",
+  "CREATE TABLE my_table (created_at timestamptz NOT NULL DEFAULT (now() AT LOCAL))",
+  "CREATE TABLE my_table (created_at timestamptz NOT NULL DEFAULT '1 hour'::interval + (current_timestamp AT TIME ZONE 'UTC'))",
```
Comment (tylergannon) [inline test/deparse_tests.c:415]:
```suggestion
  "CREATE TABLE my_table (created_at timestamptz NOT NULL DEFAULT '1 hour'::interval + (current_timestamp AT TIME ZONE 'UTC'))",
  "ALTER TABLE my_table ADD COLUMN created_at timestamptz NOT NULL DEFAULT '1 hour'::interval + (current_timestamp AT TIME ZONE 'UTC')",
```

Just to make explicit that the fix includes ADD COLUMN with DEFAULT

**Review by tylergannon — CHANGES_REQUESTED** [review]
Thanks for this!  Enjoyed reading through the commits and it helped me get a better sense of the flow through the deparser code.  True, as a first-time committer I would not have landed a correct solution like this.

Not sure if it adds any information to your tests, but I added one example to the deparser test to explicitly support ALTER TABLE ... ADD COLUMN.

**Review by msepga — APPROVED** [review]

File: test/deparse_tests.c (hunk 1 above)
Comment (lfittl) [reply test/deparse_tests.c:415]: Yep, good idea!

**Comment by lfittl** [comment]
@tylergannon Thanks again for raising this - merged!

FYI, we're currently planning to tag a libpg_query version with this (that also brings it to pg_query_go) sometime in the early new year, since there are a few other minor things we may take care of before the next release.

My understanding is you have a working solution for your project in the interim, but let us know if it'd be important for you to get an earlier patch release.
